File: backend/app/utils/batch_names.py
from app.schemas import PageInput


def format_batch_name(page_inputs: list[PageInput]) -> str:
    if not page_inputs:
        return "Unnamed batch"

    first = page_inputs[0]
    location = f"{first.city}, {first.state}" if first.state else first.city
    label = f"{first.puja} · {location}"
    extra_pages = len(page_inputs) - 1
    if extra_pages > 0:
        return f"{label} + {extra_pages} more"
    return label

File: frontend/src/utils/batchNames.ts
type PageInput = {
  puja: string;
  city: string;
  state: string;
};

export function formatBatchName(pageInputs: PageInput[]): string {
  if (pageInputs.length === 0) {
    return "Unnamed batch";
  }

  const first = pageInputs[0];
  const location = first.state ? `${first.city}, ${first.state}` : first.city;
  const label = `${first.puja} · ${location}`;
  const extraPages = pageInputs.length - 1;
  if (extraPages > 0) {
    return `${label} + ${extraPages} more`;
  }
  return label;
}
